nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteSQL.java

```diff
@@ -164,17 +166,19 @@ public void process(final OutputStream out) throws IOException {
             });
 
             // set attribute how many rows were selected
-            outgoing = session.putAttribute(outgoing, RESULT_ROW_COUNT, nrOfRows.get().toString());
+            fileToProcess = session.putAttribute(fileToProcess, RESULT_ROW_COUNT, nrOfRows.get().toString());
 
-            logger.info("{} contains {} Avro records; transferring to 'success'", new Object[] {outgoing, nrOfRows.get()});
-            session.getProvenanceReporter().modifyContent(outgoing, "Retrieved " + nrOfRows.get() + " rows", stopWatch.getElapsed(TimeUnit.MILLISECONDS));
-            session.transfer(outgoing, REL_SUCCESS);
+            logger.info("{} contains {} Avro records; transferring to 'success'",
+                    new Object[] {fileToProcess, nrOfRows.get()});
+            session.getProvenanceReporter().modifyContent(fileToProcess, "Retrieved " + nrOfRows.get() + " rows",
+                    stopWatch.getElapsed(TimeUnit.MILLISECONDS));
+            session.transfer(fileToProcess, REL_SUCCESS);
         } catch (final ProcessException | SQLException e) {
-            if (incoming == null) {
+            if (fileToProcess == null) {
                 logger.error("Unable to execute SQL select query {} due to {}. No incoming flow file to route to failure", new Object[] {selectQuery, e});
             } else {
-                logger.error("Unable to execute SQL select query {} for {} due to {}; routing to failure", new Object[] {selectQuery, incoming, e});
-                session.transfer(incoming, REL_FAILURE);
+                logger.error("Unable to execute SQL select query {} for {} due to {}; routing to failure", new Object[] {selectQuery, fileToProcess, e});
+                session.transfer(fileToProcess, REL_FAILURE);
```

Comment:
The Failure relationship states "Incoming FlowFile will be penalized and routed to this relationship". This needs to be added